src/main/java/com/minecolonies/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -0,0 +1,307 @@
+    {
+        if(!walkToBuilding())
+        {
+            final AbstractBuildingWorker workBuilding = getOwnBuilding();
+            if(workBuilding != null)
+            {
+                final TileEntityColonyBuilding chest = workBuilding.getTileEntity();
+                for (int i = 0; i < workBuilding.getTileEntity().getSizeInventory(); i++)
+                {
+                    final ItemStack stack = chest.getStackInSlot(i);
+
+                    if (stack == null)
+                    {
+                        continue;
+                    }
+
+                    if (stack.getItem() instanceof ItemArmor && worker.getItemStackFromSlot(((ItemArmor) stack.getItem()).armorType) == null)
+                    {
+                        final int emptySlot = worker.getInventoryCitizen().getFirstEmptySlot();
+                        worker.getInventoryCitizen().setInventorySlotContents(emptySlot, stack);
+                        chest.setInventorySlotContents(i, null);
+                    }
+                    else if(!(stack.getItem() instanceof ItemBow || stack.getItem() instanceof ItemSword || stack.getItem() instanceof ItemTool))
+                    {
+                    //todo dump everything which isn't a weapon
+                    }
+                }
+            }
+            attacksExecuted = 0;
+            return AIState.GUARD_SEARCH_TARGET;
+        }
+        return AIState.GUARD_RESTOCK;
+    }
+
+    /**
+     * Updates the equipment. Always take the first item of each type and set it.
+     */
+    protected void updateArmor()
+    {
+        worker.setItemStackToSlot(EntityEquipmentSlot.CHEST, null);
+        worker.setItemStackToSlot(EntityEquipmentSlot.FEET, null);
+        worker.setItemStackToSlot(EntityEquipmentSlot.HEAD, null);
+        worker.setItemStackToSlot(EntityEquipmentSlot.LEGS, null);
+
+        for(int i = 0; i < worker.getInventoryCitizen().getSizeInventory(); i++)
+        {
+            ItemStack stack = worker.getInventoryCitizen().getStackInSlot(i);
+
+            if(stack == null || stack.stackSize == 0)
+            {
+                worker.getInventoryCitizen().setInventorySlotContents(i, null);
+                continue;
+            }
+
+            if(stack.getItem() instanceof ItemArmor && worker.getItemStackFromSlot(((ItemArmor) stack.getItem()).armorType)== null)
+            {
+                worker.setItemStackToSlot(((ItemArmor) stack.getItem()).armorType, stack);
+            }
+        }
+    }
+
+    /**
+     * Chooses a target from the list.
+     * @return the next state.
+     */
+    protected AIState getTarget()
+    {
+        if(entityList.isEmpty())
+        {
+            return AIState.GUARD_PATROL;
+        }
+
+        if(entityList.get(0) instanceof EntityPlayer)
+        {
+            if(worker.getColony() != null && worker.getColony().getPermissions().getRank((EntityPlayer) entityList.get(0)) == Permissions.Rank.HOSTILE)
```

Comment:
nice!

Still I think we should have some utility method, like hatPlayerRank(player, Rank) instead of doing it with ==